### DISCUSSION TITLE
Logging input and output from Cline to the LLM's

### DISCUSSION THREAD

**Discussion by MBrekhof** [cline/cline#786] [Feature Requests]
I created a fork and used Cline to create a new version of itself that logs all communication with the LLM's. Not sure if this is of interest to others and I should create a pull request?

(https://github.com/MBrekhof/cline by the way)

**Comment by Jichao** [comment]
interested...please create a pull request.

**Comment by MBrekhof** [comment]
Because of issue 947 I found out that Cline already stores conversation in global storage (lib/app support/code/saoudrizwan.claude-dev/user/global/tasks/conversation_history.json). Please check this, if this is not sufficient I'll create a pull request

**Reply by 7UR7L3** [reply]
if anyone else is tyring to find this and isn't very familiar with vs code things, i found it's e.g. `%APPDATA%\Code\User\globalStorage\rooveterinaryinc.roo-cline\tasks\1735509949166`  that has `api_conversation_history.json` and `ui_messages.json` (i assume it's equivalent for cline proper)

**Reply by user202729** [reply]
in Linux the path could be e.g. `~/.config/Code - OSS/User/globalStorage/saoudrizwan.claude-dev/tasks/1111111111111/ui_messages.json`

**Reply by ericthomas1** [reply]
Strange...my Cline conversation history is at `C:\Users\<me>\AppData\Roaming\Code\User\globalStorage\saoudrizwan.claude-dev\tasks`. Why is `saoudrizwan.claude-dev` in this path?

**Comment by pdrago97** [comment]
Hey Guys, very hot topic. I'm trying to use cline with my company's Ollama. But I'm having 404 error on the call. If there was any log available to these set up calls, would help me a lot! Thanks.

**Reply by Berkayprof** [reply]
> Hey Guys, very hot topic. I'm trying to use cline with my company's Ollama. But I'm having 404 error on the call. If there was any log available to these set up calls, would help me a lot! Thanks.

This an problem I'm also facing. Have you yet found an solution?

**Comment by user202729** [comment]
There's a (agent-agnostic) workaround of running a litellm proxy with arize phoenix logging.
